ext/imap/php_imap.c

```diff
@@ -3053,136 +3051,141 @@ PHP_FUNCTION(imap_fetch_overview)
-						"subject", strlen("subject"),
-						env->subject
-					);
-				}
-				if (env->from) {
-					env->from->next=NULL;
-					address =_php_rfc822_write_address(env->from);
-					if (address) {
-						zend_update_property_str(
-							Z_OBJCE_P(&myoverview), Z_OBJ_P(&myoverview),
-							"from", strlen("from"),
-							address
-						);
-						zend_string_release(address);
-					}
-				}
-				if (env->to) {
-					env->to->next = NULL;
-					address = _php_rfc822_write_address(env->to);
-					if (address) {
-						zend_update_property_str(
-							Z_OBJCE_P(&myoverview), Z_OBJ_P(&myoverview),
-							"to", strlen("to"),
-							address
-						);
-						zend_string_release(address);
-					}
-				}
-				if (env->date) {
-					zend_update_property_string(
-						Z_OBJCE_P(&myoverview), Z_OBJ_P(&myoverview),
-						"date", strlen("date"),
-						(char*)env->date
-					);
-				}
-				if (env->message_id) {
-					zend_update_property_string(
-						Z_OBJCE_P(&myoverview), Z_OBJ_P(&myoverview),
-						"message_id", strlen("message_id"),
-						env->message_id
-					);
-				}
-				if (env->references) {
-					zend_update_property_string(
-						Z_OBJCE_P(&myoverview), Z_OBJ_P(&myoverview),
-						"references", strlen("references"),
-						env->references
-					);
-				}
-				if (env->in_reply_to) {
-					zend_update_property_string(
-						Z_OBJCE_P(&myoverview), Z_OBJ_P(&myoverview),
-						"in_reply_to", strlen("in_reply_to"),
-						env->in_reply_to
-					);
-				}
-				zend_update_property_long(
-					Z_OBJCE_P(&myoverview), Z_OBJ_P(&myoverview),
-					"size", strlen("size"),
-					elt->rfc822_size
-				);
-				zend_update_property_long(
-					Z_OBJCE_P(&myoverview), Z_OBJ_P(&myoverview),
-					"uid", strlen("uid"),
-					mail_uid(imap_conn_struct->imap_stream, i)
-				);
-				zend_update_property_long(
-					Z_OBJCE_P(&myoverview), Z_OBJ_P(&myoverview),
-					"msgno", strlen("msgno"),
-					i
-				);
-				zend_update_property_long(
-					Z_OBJCE_P(&myoverview), Z_OBJ_P(&myoverview),
-					"recent", strlen("recent"),
-					elt->recent
-				);
-				zend_update_property_long(
-					Z_OBJCE_P(&myoverview), Z_OBJ_P(&myoverview),
-					"flagged", strlen("flagged"),
-					elt->flagged
-				);
-				zend_update_property_long(
-					Z_OBJCE_P(&myoverview), Z_OBJ_P(&myoverview),
-					"answered", strlen("answered"),
-					elt->answered
-				);
-				zend_update_property_long(
-					Z_OBJCE_P(&myoverview), Z_OBJ_P(&myoverview),
-					"deleted", strlen("deleted"),
-					elt->deleted
-				);
-				zend_update_property_long(
-					Z_OBJCE_P(&myoverview), Z_OBJ_P(&myoverview),
-					"seen", strlen("seen"),
-					elt->seen
-				);
-				zend_update_property_long(
+	if (!status) {
+		RETURN_EMPTY_ARRAY();
+	}
+
+	array_init(return_value);
+	for (unsigned long  i = 1; i <= imap_conn_struct->imap_stream->nmsgs; i++) {
+		MESSAGECACHE *elt = mail_elt(imap_conn_struct->imap_stream, i);
+		ENVELOPE *env = mail_fetch_structure(imap_conn_struct->imap_stream, i, NIL, NIL);
```

Comment:
There's a subtle semantic change here.
In the previous version of this code, it only did the `mail_fetch_structure` call when `mail_elt` succeeded. I think it's safest if we keep the same semantics.